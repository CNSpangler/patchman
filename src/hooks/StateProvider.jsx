import React, { createContext, useContext, useEffect, useState } from 'react';
import PropTypes from 'prop-types';

const StateProvider = ({ children }) => {
  const [url, setUrl] = useState('');
  const [body, setBody] = useState('');
  const [method, setMethod] = useState('GET');
  const [results, setResults] = useState('');
  const [history, setHistory] = useState([]);

  const handleUrlChange = ({ target }) => {
    setUrl(target.value);
  };
  
  const handleMethodChange = ({ target }) => {
    setMethod(target.value);
  };
  
  const handleBodyChange = ({ target }) => {
    setBody(target.value);
  };
  
  const handleSubmit = (e) => {
    e.preventDefault();
  
    makeRequest(url, method, body)
      .then(response => setResults(response));
  
    // dependent state change
    setHistory(prevHistory => ([
      ...prevHistory,
      {
        url,
        method,
        body
      }
    ]));
  };

  return (
    <StateContext.Provider value={{ url, body, method, results, history, handleUrlChange, handleMethodChange, handleBodyChange, handleSubmit }}>
      {children}
    </StateContext.Provider>
  );
};

StateProvider.propTypes = {
  children: PropTypes.node
};

export default StateProvider;
