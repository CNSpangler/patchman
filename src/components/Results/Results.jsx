import React from 'react';
import ReactJson from 'react-json-view';
import styles from '../App/App.css';
import { useResults } from '../../hooks/StateProvider';

const Results = () => {
  const results = useResults();

  return (
    <div className={styles.Results}>
      {results ? <ReactJson src={results} /> : <div>Please make a request above</div>}
    </div>
  );
};

export default Results;
