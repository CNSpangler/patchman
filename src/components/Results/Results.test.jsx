import React from 'react';
import { shallow } from 'enzyme';
import Results from './Results.jsx';
import { StateProvider } from '../../hooks/StateProvider.jsx';

describe('Results component', () => {
  it('renders Results ', () => {
    const wrapper = shallow(
      <StateProvider>  
        <Results />
      </StateProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
