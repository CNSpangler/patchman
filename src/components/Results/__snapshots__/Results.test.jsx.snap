// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component renders Results  1`] = `
<ContextProvider
  value={
    Object {
      "body": "",
      "handleBodyChange": [Function],
      "handleMethodChange": [Function],
      "handleSubmit": [Function],
      "handleUrlChange": [Function],
      "history": Array [],
      "method": "GET",
      "results": "",
      "url": "",
    }
  }
>
  <Results />
</ContextProvider>
`;
